src/test/java/hudson/plugins/git/GitChangeSetTest.java

```diff
@@ -41,6 +44,119 @@ public void testFindOrCreateUser() {
+        assertThat(user, is(User.getUnknown()));
+    }
+
+    @Test
+    public void testFindOrCreateUserEmptyAuthorEmail() {
+        final GitChangeSet changeset = GitChangeSetUtil.genChangeSet(random.nextBoolean(), random.nextBoolean());
+        final boolean createAccountBasedOnEmail = true;
+        final boolean useExistingAccountBasedOnEmail = random.nextBoolean();
+        final String csAuthor = "babbage";
+        final String csAuthorEmail = "";
+        User user = changeset.findOrCreateUser(csAuthor, csAuthorEmail, createAccountBasedOnEmail, useExistingAccountBasedOnEmail);
+        assertThat(user, is(User.getUnknown()));
+    }
+
+    @Test
+    public void testFindOrCreateUserEmptyAuthorEmailDoNotCreateAccountBasedOnEmail() {
+        final GitChangeSet changeset = GitChangeSetUtil.genChangeSet(random.nextBoolean(), random.nextBoolean());
+        final boolean createAccountBasedOnEmail = false;
+        final boolean useExistingAccountBasedOnEmail = random.nextBoolean();
+        final String csAuthor = "babbage-do-not-create";
+        final String csAuthorEmail = "";
+        User user = changeset.findOrCreateUser(csAuthor, csAuthorEmail, createAccountBasedOnEmail, useExistingAccountBasedOnEmail);
+        assertThat(user, is(User.getUnknown()));
+    }
+
+    @Test
+    public void testFindOrCreateUserEmptyAuthorDoNotCreateAccountBasedOnEmail() {
+        final GitChangeSet changeset = GitChangeSetUtil.genChangeSet(random.nextBoolean(), random.nextBoolean());
+        final boolean createAccountBasedOnEmail = false;
+        final boolean useExistingAccountBasedOnEmail = random.nextBoolean();
+        final String csAuthor = "";
+        final String csAuthorEmail = "babbage-empty-author@example.com";
+        User user = changeset.findOrCreateUser(csAuthor, csAuthorEmail, createAccountBasedOnEmail, useExistingAccountBasedOnEmail);
+        assertThat(user, is(User.getUnknown()));
+    }
+
+    @Test
+    public void testFindOrCreateUserBadAuthorEmailDoNotCreateAccountBasedOnEmail() {
+        final GitChangeSet changeset = GitChangeSetUtil.genChangeSet(random.nextBoolean(), random.nextBoolean());
+        final boolean createAccountBasedOnEmail = false;
+        final boolean useExistingAccountBasedOnEmail = random.nextBoolean();
+        final String csAuthor = "babbage-do-not-create";
+        final String csAuthorEmail = "@";
+        User user = changeset.findOrCreateUser(csAuthor, csAuthorEmail, createAccountBasedOnEmail, useExistingAccountBasedOnEmail);
+        assertThat(user, is(User.getUnknown()));
+    }
+
+    @Test
+    public void testFindOrCreateUserOKAuthorEmailDoNotCreateAccountBasedOnEmail() {
+        final GitChangeSet changeset = GitChangeSetUtil.genChangeSet(random.nextBoolean(), random.nextBoolean());
+        final boolean createAccountBasedOnEmail = false;
+        final boolean useExistingAccountBasedOnEmail = random.nextBoolean();
+        final String csAuthor = "babbage-do-not-create-either";
```

Comment:
Make the author string match the results of the test for easier reading
```suggestion
        final String csAuthor = "babbage-will-be-created";
```